datalad_dataverse/remote.py

```diff
@@ -105,15 +105,13 @@ class DataverseRemote(ExportRemote, BaseDataverseRemote):
     # Export API
     #
     def checkpresentexport(self, key, remote_file):
-        stored_id = self._get_annex_fileid_record(key)
-        if stored_id is not None:
-            # Only check latest version in export mode. Doesn't currently
-            # work for keys from older versions, since annex fails to even
-            # try. See https://github.com/datalad/datalad-dataverse/issues/146#issuecomment-1214409351.
-            return self._dvds.has_fileid_in_latest_version(stored_id)
+        remote_file = mangle_path(remote_file)
+        stored_ids = self._get_annex_fileid_record(key)
+        if stored_ids:
+            # Only check latest version in export mode.
+            return self._get_fileid_from_remotepath(
+                remote_file, latest_only=True) in stored_ids
         else:
-            # In export mode, we need to fix remote paths:
-            remote_file = mangle_path(remote_file)
             return self._dvds.has_path_in_latest_version(remote_file)
```

Comment:
Trying to understand in order to place a comment to help with that here:

- if IDs are stored: we check whether the current ID for the remote path matches any of them
- if IDs are not stored: we are happy with anything at that path

what is the rational for this difference?

Reply:
Didn't touch that aspect in this PR. I think the fallback to plain path matching is questionable now indeed.
I think this is largely a left over from when there was no ID on record and/or when the logic was shared with annex mode.
There was one more reason originally: Being able to deal with datalad-annex' keys. When those are retrieved, there is no repo yet, that could possibly have the ID on record. I'm not sure about this aspect ATM, which is why I left it in place.

Reply:
Thx, I added comment accordingly